docs/migration_guide_v2.md

```diff
@@ -0,0 +1,647 @@
+  * [Cleanup of old and unmaintained dependencies](#cleanup-of-old-and-unmaintained-dependencies)
+  * [New tests infrastructure](#new-tests-infrastructure)
+* [Commands Migration](#commands-migration)
+  * [`scw help`](#scw-help)
+  * [`scw attach`](#scw-attach)
+  * [`scw commit`](#scw-commit)
+  * [`scw cp`](#scw-cp)
+  * [`scw create`](#scw-create)
+  * [`scw events`](#scw-events)
+  * [`scw exec`](#scw-exec)
+  * [`scw history`](#scw-history)
+  * [`scw images`](#scw-images)
+  * [`scw info`](#scw-info)
+  * [`scw inspect`](#scw-inspect)
+  * [`scw kill`](#scw-kill)
+  * [`scw login`](#scw-login)
+  * [`scw logout`](#scw-logout)
+  * [`scw logs`](#scw-logs)
+  * [`scw port`](#scw-port)
+  * [`scw products`](#scw-products)
+  * [`scw ps`](#scw-ps)
+  * [`scw rename`](#scw-rename)
+  * [`scw restart`](#scw-restart)
+  * [`scw rm`](#scw-rm)
+  * [`scw rmi`](#scw-rmi)
+  * [`scw run`](#scw-run)
+  * [`scw search`](#scw-search)
+  * [`scw start`](#scw-start)
+  * [`scw stop`](#scw-stop)
+  * [`scw tag`](#scw-tag)
+  * [`scw top`](#scw-top)
+  * [`scw version`](#scw-version)
+  * [`scw wait`](#scw-wait)
+
+## Justification for a v2
+
+### Multiple products design
+
+CLI v1 was created while targeting a single Scaleway Elements product: instance.
+Scaleway got many more products in its Elements ecosystem that need to be available from the CLI.
+Having ad-hoc commands for all our products was not going to be a scalable solution.
+We needed to have a more systematic approach across all products.
+
+In CLI v2, we have a common syntax across all our products: service sub-resource verb.
+As an example:
+
+```shell
+# v2
+scw instance server list
+```
+
+* **instance**: Refers to the product namespace
+* **server**: Refers to a resource maintained in the product namespace
+* **list**: Refers to a verb apply to the currently selected API
+
+In English, it would be: "list all servers available on the instance API".
+
+### Consistent workflow across products
+
+CLI v1 was designed to offer a syntax close to the Docker syntax.
+For instance, running a command such as `echo foobar` on a remote server is `scw run ubuntu-bionic echo foobar` which mimic `docker run ubuntu echo foobar`.
+While this can be useful for some tasks, there are plenty of actions that do not fit in with this paradigm.
+E.g., attaching to a running server sub-resources such as volumes or security groups are not performed easily using this paradigm because Docker doesn't provide the same features.
+
+In CLI v2 we offer support for a wide range of actions on all resources present and coming in the Scaleway Elements ecosystem.
+Actions are organized around a set of verbs such as `list`, `get`, `create`, `update` that can be used with a wide variety of products and do not suppose any preconceived workflow.
+
+### Automated code generation
+
+CLI v1 did not have any code generation features to easily create supports in all SDKs and developer tools we support.
```

Comment:
It's not the CLI that has code generation feature. 

It's code generation that is used to build the CLI, and also the SDKs and other dev tools. 

`CLI v1 did not take advantage of code generation features to easily create supports like in SDKs and other developper tools`